src/uct/ib/dc/base/dc_iface.c

```diff
@@ -545,7 +546,9 @@ void uct_dc_handle_failure(uct_ib_iface_t *ib_iface, uint32_t qp_num,
     uct_dc_iface_dci_put(iface, dci);
     ucs_assert_always(ep->dci == UCT_DC_EP_NO_DCI);
 
-    iface->super.super.ops->set_ep_failed(ib_iface, &ep->super.super, status);
+    ep_status = iface->super.super.ops->set_ep_failed(ib_iface,
+                                                      &ep->super.super, status);
+    ucs_assert_always(ep_status == UCS_OK);
```

Comment:
need that all transports will do same as rc_mlx5 - instead of printing assertion, print the original error with fatal log level